hw_i2c/sample-implementations/zephyr_user_space/sensirion_hw_i2c_implementation.c

```diff
@@ -0,0 +1,126 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <zephyr.h>
+#include <unistd.h>
+#include <drivers/i2c.h>
+#include <device.h>
+#include <string.h>
+
+#include "sensirion_arch_config.h"
+#include "sensirion_i2c.h"
+#include "sensirion_common.h"
+
+/* SPS30 I2C device. */
+static struct device *sps30_i2c_dev;
+
+/**
+ * Select the current i2c bus by index.
+ * All following i2c operations will be directed at that bus.
+ *
+ * @param bus_idx   Bus index to select
+ * @returns         0 on success, an error code otherwise
+ */
+int16_t sensirion_i2c_select_bus(uint8_t bus_idx)
+{
+	int16_t status = STATUS_FAIL;
+	char bus_name[6] = "I2C_0";
+
+	/* Get device from the provided bus index (only support 0-9). */
+	bus_name[4] = bus_idx +'0';
+	sps30_i2c_dev = device_get_binding(bus_name);
+	if (sps30_i2c_dev != NULL)
+		status = STATUS_OK;
+
+	return (status);
+}
+
+/**
+ * Initialize all hard- and software components that are needed for the I2C
+ * communication.
+ */
+void sensirion_i2c_init(void)
+{
+	/* Device (specified by sps30_i2c_dev) is already initialized by the Zephyr
+	 * boot-up process. Nothing to be done here. */
+}
+
+/**
+ * Release all resources initialized by sensirion_i2c_init().
+ */
+void sensirion_i2c_release(void)
+{
+	sps30_i2c_dev = NULL;
+}
+
+/**
+ * Execute one read transaction on the I2C bus, reading a given number of bytes.
+ * If the device does not acknowledge the read command, an error shall be
+ * returned.
+ *
+ * @param address 7-bit I2C address to read from
+ * @param data    pointer to the buffer where the data is to be stored
+ * @param count   number of bytes to read from I2C and store in the buffer
+ * @returns 0 on success, error code otherwise
+ */
+int8_t sensirion_i2c_read(uint8_t address, uint8_t *data, uint16_t count)
+{
+	return i2c_read(sps30_i2c_dev, data, count, address);
+}
+
+/**
+ * Execute one write transaction on the I2C bus, sending a given number of
+ * bytes. The bytes in the supplied buffer must be sent to the given address. If
+ * the slave device does not acknowledge any of the bytes, an error shall be
+ * returned.
+ *
+ * @param address 7-bit I2C address to write to
+ * @param data    pointer to the buffer containing the data to write
+ * @param count   number of bytes to read from the buffer and send over I2C
+ * @returns 0 on success, error code otherwise
+ */
+int8_t sensirion_i2c_write(uint8_t address, const uint8_t *data,
+						   uint16_t count)
```

Comment:
Could you change the whole file's indentation to using spaces to match the repo's style? (You should be able to have the file reformatted with `clang-format -i $(find . -name "*.[ch]")`